### PR TITLE
- Add DS treatment in case of SHA-1 in DNSSEC01

### PR DESCRIPTION
**Pull request by vlevigneron**
fixes zonemaster/zonemaster-engine#712

### CODE REVIEW

**Comment by sandoche2k** [comment]
@matsduf and @mattias-p please review

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 1]
```diff
@@ -541,6 +542,10 @@ Readonly my %TAG_DESCRIPTIONS => (
         __x    # DNSSEC:DS_ALGORITHM_RESERVED
           "{ns}/{address} returned a DS record created by with an algorithm not assigned (algorithm number {algorithm_number}), which is not OK. The DS record is for the DNSKEY record with keytag {keytag} in zone {zone}.", @_;
     },
+    DS_ALGO_SHA1_DEPRECATED => sub {
+        __x    # DNSSEC:DS_ALGO_SHA1_DEPRECATED
+          "{ns}/{address} returned a DS record created by algorithm {algorithm_number} ({algorithm_mnemonic}), which is, while it is still widely used, deprecated. The DS record is for the DNSKEY record with keytag {keytag} in zone {zone}.", @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:547]: I suggest shorter lines in Perl module to make it easier to inspect and read. And a somewhat updated message ID. I think we should be more explicit in our message IDs.

```
"Nameserver {ns}/{address} returned a DS record created by algorithm {algorithm_number}  "  . 
"({algorithm_mnemonic}) which is deprecated, while it is still widely used. The DS record is " . 
"for the DNSKEY record with keytag {keytag} in zone {zone}.", @_;

Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm:547]: OK, i'm going to shorten all lines of DNSSEC messages.

**Comment by matsduf** [comment]
@vlevigneron, did you see my comment on en.po?

**Comment by vlevigneron** [comment]
> @vlevigneron, did you see my comment on en.po?

Yes. How can I cleanly remove the file from this commit ?

**Comment by matsduf** [comment]
I accept a force-push to the pull request.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 2]
```diff
@@ -430,191 +430,224 @@ sub metadata {
...
     },
     HAS_NSEC => sub {
         __x    # DNSSEC:HAS_NSEC
-          "The zone has NSEC records.", @_;
+          'The zone has NSEC records.', @_;
     },
     INCONSISTENT_DNSSEC => sub {
         __x    # DNSSEC:INCONSISTENT_DNSSEC
-          "Some, but not all, nameservers for zone {zone} respond with neither NSEC nor NSEC3 records when such records are expected.", @_;
+          'Some, but not all, nameservers for zone {zone} respond with neither NSEC nor NSEC3 records when such records are expected.', @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:642]: Still long line.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 3]
```diff
@@ -430,191 +430,224 @@ sub metadata {
...
-          "Nameserver {ns}/{address} responded with no RRSIG for RRset {rrtype} created by the algorithm {algorithm}.", @_;
+          'Nameserver {ns}/{address} responded with no RRSIG for RRset {rrtype} created by the '
+          . 'algorithm {algorithm}.',
+          @_;
     },
     ALL_ALGO_SIGNED => sub {
-       __x    # DNSSEC:ALL_ALGO_SIGNED
-          "All the tested RRset (SOA/DNSKEY/NS) are signed by each algorithm present in the DNSKEY RRset", @_;
+        __x    # DNSSEC:ALL_ALGO_SIGNED
+          'All the tested RRset (SOA/DNSKEY/NS) are signed by each algorithm present in the DNSKEY RRset',
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:477]: "." is missing at the end of the line/message.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm [hunk 4]
```diff
@@ -430,191 +430,224 @@ sub metadata {
...
     },
     INCONSISTENT_DNSSEC => sub {
         __x    # DNSSEC:INCONSISTENT_DNSSEC
-          "Some, but not all, nameservers for zone {zone} respond with neither NSEC nor NSEC3 records when such records are expected.", @_;
+          'Some, but not all, nameservers for zone {zone} respond with neither NSEC nor NSEC3 records when such records are expected.', @_;
     },
     INCONSISTENT_NSEC_NSEC3 => sub {
         __x    # DNSSEC:INCONSISTENT_NSEC_NSEC3
-          "Some nameservers for zone {zone} respond with NSEC records and others respond with NSEC3 records. Consistency is expected.", @_;
+          'Some nameservers for zone {zone} respond with NSEC records and others respond with NSEC3 records. Consistency is expected.', @_;
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/DNSSEC.pm:646]: Still a long line.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 3 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm:477]: yes some messages have typos, I fixed none of them because it has on impact on existing  *.po files

Comment (matsduf) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm:477]: OK.

File: lib/Zonemaster/Engine/Test/DNSSEC.pm (hunk 2 above)
Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm:642]: I agree, this was just a first pass of homogenization of messages (replace all double quotes for instance). I plan to make a more deep cleaning after the release.

Comment (matsduf) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm:642]: You shortened all long msgid but two. Why not the last two?

Comment (vlevigneron) [reply lib/Zonemaster/Engine/Test/DNSSEC.pm]: I shortened these 2 messages

**Pull request merged**